src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -84,12 +84,6 @@ class ObjectWaiter : public CHeapObj<mtThread> {
 //
 // ObjectMonitor Layout Overview/Highlights/Restrictions:
 //
-// - The _metadata field must be at offset 0 because the displaced header
-//   from markWord is stored there. We do not want markWord.hpp to include
-//   ObjectMonitor.hpp to avoid exposing ObjectMonitor everywhere. This
-//   means that ObjectMonitor cannot inherit from any other class nor can
-//   it use any virtual member functions. This restriction is critical to
-//   the proper functioning of the VM.
```

Comment:
So having it at offset zero is no longer a restriction but we still keep it there because of performance. That in turn means we still can't inherit from any other class, so that is still a restriction. Suggestion:
```
// For performance reasons we ensure the _metadata field is located at offset 0, which in turn
// means that ObjectMonitor can't inherit from any other class nor use any virtual member functions.
```

Reply:
Thanks. I think for contended monitors this could be important - it always was in the past.